packages/core-data/src/actions.js

```diff
@@ -406,14 +407,14 @@ export const editEntityRecord =
 export const undo =
 	() =>
 	( { select, dispatch } ) => {
-		const undoEdit = select.getUndoEdit();
+		// Todo: we shouldn't have to pass "root" here.
+		const undoEdit = select( ( state ) => getUndoEdits( state.root ) );
```

Comment:
@adamziel @jsnajdr I've learned today that:

 - you can call private selectors like that in "thunks"
 - you can't (or at least I didn't see any way to do it) call `unlock( select ).somePrivateSelector()` in "thunks".
 - The first approach is buggy because I had to pass `state.root` instead of just `state` (`state.root` should be just an internal thing, the store author shouldn't even know it exists).

Fixing that bug is going to be a breaking change. Maybe we can still "proxy" .root and deprecate it somehow though. I'm not fixing this in this unrelated PR but just wanted to let you know about this.

Reply:
> If select doesn't expose private selectors, we need to fix the thunkArgs construction: instead of getSelectors() we need to get the "unlocked" selectors.

Yes, I tried without unlocking and it doesn't work, these things are not available.

> It seems like this was intentional because we're passing store.__unstableOriginalGetState() to select() [here](https://github.com/WordPress/gutenberg/blob/2b687d03ecdfb52229c25d5d455038b14fc8fa89/packages/data/src/redux-store/index.js#L159). If we didn't want that, we could very easily pass store.getState(). Or was this an oversight that I and @adamziel didn't catch 2 years ago?

I'd argue that it's oversight because all selectors receive the "state" and not the "original state" with metadata (which is an internal thing added by the data module).